src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -82,9 +92,14 @@ private static boolean isValidPath(Path filePath) {
      * Saves all data to this storage file.
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws FileDeletedException if the created storage file has been deleted while the program is running
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook) throws StorageOperationException, FileDeletedException {
 
+    	File storageFile = path.toFile();
+    	if (!storageFile.exists() && isStorageFileCreated) {
+    		throw new FileDeletedException("File does not exist, it might have been deleted.");
```

Comment:
as discussed in the tutorial, this may not be the most apt way to handle the exception; in this case I believe the program halts. what you can do is go ahead and create the file anyway, and use the exception to inform user of the same.